aca_required_temp.py

```diff
@@ -0,0 +1,189 @@
+        # That will give us the closest-to-nominal best choice
+        for roll in [nom_roll - rolldiff, nom_roll + rolldiff]:
+            roll_t_ccd = max_temp(ra, dec, roll, time=time, cone_stars=cone_stars)
+            if roll_t_ccd is not None:
+                if best_t_ccd is None or roll_t_ccd > best_t_ccd:
+                    best_t_ccd = roll_t_ccd
+                    best_roll = roll
+                if best_t_ccd == WARM_T_CCD:
+                    break
+    return best_t_ccd, best_roll
+
+
+def temps_for_attitude(ra, dec, start='2014-09-01', stop='2015-12-31'):
+    # reset the caches at every new attitude
+    global TEMP_CACHE
+    TEMP_CACHE.clear()
+
+    # set the agasc lookup time to be in the middle of the cycle for
+    # proper motion correction
+    agasc_mid_time = DateTime((DateTime(start).secs + DateTime(stop).secs)
+                              / 2).date
+
+    # Get stars in this field
+    cone_stars = get_agasc_cone(ra, dec, time=agasc_mid_time)
+
+    # get a list of days
+    start = DateTime(start)
+    stop = DateTime(stop)
+    days = start + np.arange(stop - start)
+
+    temps = {}
+    # loop over them
+    for day in days.date:
+        nom_roll = nominal_roll(ra, dec, time=day)
+        day_pitch = pitch(ra, dec, time=day)
+        if day_pitch < 45 or day_pitch > 170:
+            continue
+
+        nom_roll_t_ccd = max_temp(ra, dec, nom_roll, time=day, cone_stars=cone_stars)
+        # if we can get the nominal roll catalog at warmest temp, why check the rest?
+        if nom_roll_t_ccd == WARM_T_CCD:
+            best_t_ccd = nom_roll_t_ccd
+            best_roll = nom_roll
+        else:
+            best_t_ccd, best_roll = best_temp_roll(ra, dec, nom_roll,
+                                                   day_pitch,
+                                                   time=day,
+                                                   cone_stars=cone_stars)
+        # should we have values or skip entries for None here?
+        if best_t_ccd is None:
+            continue
+        temps["{}".format(day[0:8])] = {
+            'day': day,
+            'pitch': "{:.2f}".format(day_pitch),
+            'nom_roll': "{:.2f}".format(nom_roll),
+            'nom_roll_t_ccd': "{:.2f}".format(nom_roll_t_ccd),
+            'best_roll': "{:.2f}".format(best_roll),
+            'best_t_ccd': "{:.2f}".format(best_t_ccd)}
+    table = Table(temps.values())['day', 'pitch',
+                                  'nom_roll', 'nom_roll_t_ccd',
+                                  'best_roll', 'best_t_ccd']
+
+    table.sort('day')
+    return table
+
+def main():
+    """
+    Determine required ACA temperature for an attitude over a time range
+    """
+    opt = get_options()
+    temp_table = temps_for_attitude(opt.ra, opt.dec,
+                                    start=opt.start,
+                                    stop=opt.stop)
+    temp_table.write(opt.out, format='ascii.fixed_width_two_line')
```

Comment:
We should ask SOT MP what they want.  Fixed width two line is nice to read for humans, and easy to read in Python with io.ascii, but requires special handling for any other platform to machine-read because of the extra line.